code/modules/wiremod/shell/airlock.dm

```diff
@@ -30,6 +30,11 @@
 /obj/machinery/door/airlock/shell/canAIHack(mob/user)
 	return FALSE
 
+/obj/machinery/door/airlock/shell/allowed(mob/user)
+	if(SEND_SIGNAL(src, COMSIG_OBJ_ALLOWED, user) & COMPONENT_OBJ_ALLOW)
```

Comment:
this is bad again, you should not be calling the same signal in multiple places

Reply:
Yeah I considered this issue, I'll split this PR into two for now as I am really having trouble finding a method that can be overrided here without affecting functionality.

Reply:
use another signal for airlock shells, register both COSMIG_OBJ_ALLOWED and that new signal

Reply:
Actually COMSIG_OBJ_ALLOWED doesn't seem to be used anywhere so should it be kept or not